### PR TITLE
fix(layout): updates to flexbox styles, directives

### LINKED ISSUE #4482: Allow flex-order to be negative
According to [CSS Flexible Box Layout Module Level 1](http://www.w3.org/TR/css3-flexbox/#order-property) and [CSS Values and Units Module Level 3](http://www.w3.org/TR/css-values/#integers), the `order` property can be negative, -9 to -1, as well as positive, 1 to 9, and zero.

With this feature, developers can use positive and negative flex-orders for general (search results) and special (ads) flex items respectively.

Here's a demo based on v0.11.0-rc2 with a simple patch for this feature request:
http://codepen.io/flowersinthesand/pen/ojgrmj

### PR DESCRIPTION
**Pull request by ThomasBurleson**
* allow flex-order to be negative
* add support to observe interpolated layout values.
*  restore max-width for layouts
  *  restore `layout > flex` specifiers of max-width or max-height.
  * publish `$$mdLayout` service to allow Attribute selectors to be removed when the translation inject to Class selector finishes.
*  add `md-css-only` feature , 
  *  support use of `<body class="md-css-only">`  to use Layout features **without** any JS requirements or `ngMaterial` module dependencies.
    *  May use standard Attribute selectors
    *  May use class selectors
    *  Disables Layout directive postlinks for optimization

According to CSS Flexible Box Layout Module Level 1 and CSS Values and Units Module Level 3, the order property can be negative, -9 to -1, as well as positive, 1 to 9, and zero.

Fixes #4482. Closes #4592.

> Thx @flowersinthesand for original PR

### CODE REVIEW

**Comment by ThomasBurleson** [comment]
@topherfangio - can you review?

File: src/core/services/layout/layout.js [hunk 1]
```diff
@@ -42,25 +44,41 @@
      */
     angular.module('material.core.layout', [ 'ng' ])
 
+      /**
+       * Model of flags used by the Layout directives
+       * Allows changes while running tests or runtime app changes
+       */
+      .factory("$$mdLayout", function() {
```
Comment (ThomasBurleson) [inline src/core/services/layout/layout.js]: Should probably be a provider to allow config changes but not runtime changes.

File: src/core/services/layout/layout.js [hunk 2]
```diff
@@ -106,21 +118,36 @@
...
       .directive('showGtSm'            , attributeWithoutValue('show-gt-sm')         )
-      .directive('showLtMd'            , warnAttrNotSupported ('show-lt-md')         )
       .directive('showMd'              , attributeWithoutValue('show-md')            )
       .directive('showGtMd'            , attributeWithoutValue('show-gt-md')         )
-      .directive('showLtLg'            , warnAttrNotSupported ('show-lt-lg')         )
       .directive('showLg'              , attributeWithoutValue('show-lg')            )
-      .directive('showGtLg'            , attributeWithoutValue('show-gt-lg')         );
+      .directive('showGtLg'            , attributeWithoutValue('show-gt-lg')         )
+
+      // !! Deprecated attributes: use the `-lt` (aka less-than) notations
```
Comment (ThomasBurleson) [inline src/core/services/layout/layout.js]: Provides `$log.warn( )` details useful for migration

File: src/core/services/layout/layout.js [hunk 3]
```diff
@@ -239,4 +290,35 @@
 
     }
 
+    /**
```
Comment (ThomasBurleson) [inline src/core/services/layout/layout.js]: API addition:

*  Allows use of Attribute selectors for browsers that do not suffer performance degradations
*  Allows use of ngMaterial Layout/Flexbox CSS **without** loading the `angular-material.js` source.

File: src/core/services/layout/layout.spec.js [hunk 4]
```diff
@@ -111,4 +123,32 @@ describe('layout directives', function() {
...
+      });
+
+      expect(element.hasClass('flex-gt-md-32')).toBe(true);
+
+      scope.$apply(function() {
+        scope.size = "fishCheeks";
+      });
+
+      expect(element.hasClass('flex-gt-md-32')).toBe(false);
+      expect(element.hasClass('flex-gt-md-fishCheeks')).toBe(true);
```
Comment (ThomasBurleson) [inline src/core/services/layout/layout.spec.js]: Maybe we should check the validity of the value to see if the value (for the attribute) is an acceptable, known value; warn if not?

File: src/core/services/layout/layout.scss [hunk 5]
```diff
@@ -19,71 +19,90 @@
...
 }
 @-moz-document url-prefix() {
-  .layout-fill {
+  .layout-fill, .md-css-only [layout-fill] {
     margin: 0;
     width: 100%;
-    min-height: auto;
-    height: inherit;
+    min-height: 100%;
+    height: 100%;
```
Comment (topherfangio) [inline src/core/services/layout/layout.scss]: I don't think it needs a comment, but can you briefly explain why these were changed from auto/inherit to 100%?

File: src/core/services/layout/layout.scss [hunk 6]
```diff
@@ -154,28 +178,42 @@
...
+  .layout-row, .md-css-only [layout="row"] {
+    > .#{$flexName}-33, > .#{$flexName}-34 { max-width: 33%; }
+    > .#{$flexName}-66, > .#{$flexName}-67 { max-width: 67%;  }
   }
-  .layout-column {
-    > .#{$flexName}-33, > .#{$flexName}-34 {  }
-    > .#{$flexName}-66, > .#{$flexName}-67 {  }
+  .layout-column, .md-css-only [layout="column"] {
+    > .#{$flexName}-33, > .#{$flexName}-34 { max-height: 33%;  }
+    > .#{$flexName}-66, > .#{$flexName}-67 { max-height: 33%; }
```
Comment (topherfangio) [inline src/core/services/layout/layout.scss]: Shouldn't this be `max-height: 67%`? Copy/paste issue?

File: src/core/services/layout/layout.spec.js (hunk 4 above)
Comment (topherfangio) [inline src/core/services/layout/layout.spec.js]: I'm not so sure we should check. If someone wants to add to "extend" our CSS with their own custom sizes, it seems like that would generate a lot of warnings. Also, I think if it doesn't match, then they will have a visual warning by the fact that their site's layout is broken :-)

**Comment by topherfangio** [comment]
LGTM aside from my comments.

**Comment by ThomasBurleson** [comment]
Closed. See new PR #4760

**Pull request closed without merging**